src/routes/spirit-board/index.svelte

```diff
@@ -147,211 +147,187 @@
-        board.appendChild(newArtistElement);
-      }
+  function generateHTML(spiritBoard) {
+    const fragment = new DocumentFragment();
 
-      //Set Special Rules
-      const specialRulesContainer =
-        frame.contentDocument.querySelectorAll("special-rules-container")[0];
-      if (specialRulesContainer) {
-        specialRulesContainer.textContent = ""; // (easiest to start fresh each time)
-        let specialRulesHeader = frame.contentDocument.createElement("section-title");
-        specialRulesHeader.textContent = "SPECIAL RULES";
-        specialRulesContainer.appendChild(specialRulesHeader);
-      }
-      spiritBoard.specialRules.rules.forEach((rule) => {
-        let newRuleName = frame.contentDocument.createElement("special-rules-subtitle");
-        newRuleName.textContent = rule.name;
-        let newRuleEffect = frame.contentDocument.createElement("special-rule");
-        newRuleEffect.innerHTML = rule.effect;
-        specialRulesContainer.appendChild(newRuleName);
-        specialRulesContainer.appendChild(newRuleEffect);
-      });
+    const board = document.createElement("board");
+    fragment.append(board);
+    board.setAttribute("spirit-image", spiritBoard.nameAndArt.artPath);
+    board.setAttribute("spirit-image-scale", spiritBoard.nameAndArt.artScale);
+    board.setAttribute("spirit-border", spiritBoard.nameAndArt.bannerPath);
 
-      //Set Growth
-      const growthContainer = frame.contentDocument.querySelectorAll("growth")[0];
-      if (growthContainer) {
-        growthContainer.textContent = ""; //(easiest to start fresh each time)
-      }
-      if (!spiritBoard.growth.useGrowthSets) {
-        growthContainer.setAttribute("title", `Growth (${spiritBoard.growth.directions})`);
+    //Set Spirit Name and Image
+    const spiritName = document.createElement("spirit-name");
+    if (spiritName) {
+      spiritName.textContent = spiritBoard.nameAndArt.name;
+    }
+    board.appendChild(spiritName);
+
+    const artistName = document.createElement("artist-name");
+    artistName.textContent = spiritBoard.nameAndArt.artistCredit;
+    board.appendChild(artistName);
+
+    //Set Special Rules
+    const specialRulesContainer = document.createElement("special-rules-container");
+    board.appendChild(specialRulesContainer);
+
+    let specialRulesHeader = document.createElement("section-title");
+    specialRulesHeader.textContent = "SPECIAL RULES";
+    specialRulesContainer.appendChild(specialRulesHeader);
+    spiritBoard.specialRules.rules.forEach((rule) => {
+      let newRuleName = document.createElement("special-rules-subtitle");
+      newRuleName.textContent = rule.name;
+      let newRuleEffect = document.createElement("special-rule");
+      newRuleEffect.innerHTML = rule.effect;
+      specialRulesContainer.appendChild(newRuleName);
+      specialRulesContainer.appendChild(newRuleEffect);
+    });
+
+    const right = document.createElement("right");
+    board.appendChild(right);
```

Comment:
Previously, we had a blank HTML template spirit to work from, so we didn't have to add a <right> element. Now we're building from the ground up so we add everything. Is that right? Just making sure I understand. This function now builds a fresh HTML fragment that looks very similar to what used to live in 'holder-frame', right?

Reply:
Yes, exactly.